### PR TITLE
TINY-12729: Add async support to Fun module in Katamari 

### PR DESCRIPTION
**Pull request by soritaheng**
Related Ticket: TINY-12729

Description of Changes:
* Added async functions `constantAsync`, `identityAsync`, `neverAsync` and `alwaysAsync` to the `Fun` module
* Added tests for the new functions 

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

## Summary by CodeRabbit

* **New Features**
  * Added four asynchronous utility functions that return Promises, providing promise-based alternatives for common functional helpers to simplify async/await usage and composition.

* **Tests**
  * Added unit and property-based tests covering primitives, objects, arrays, undefined, and edge cases to validate async utility behavior.

### CODE REVIEW

File: .changes/unreleased/katamari-TINY-12729-2026-02-09.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+project: katamari
+kind: Added
+body: Added async functions `constantAsync`, `identityAsync`, `neverAsync` and `alwaysAsync` to the `Fun` module.
```
Comment (kimwoodfield) [inline .changes/unreleased/katamari-TINY-12729-2026-02-09.yaml]:
```suggestion
body: Async support to the `Fun` module via `constantAsync`, `identityAsync`, `neverAsync` and `alwaysAsync` functions.
```

Maybe the wording can be improved. No need to include the same word in `body` from `kind` (e.g. `Added`) since we've already described it.

**Comment by kimwoodfield** [comment]
Just a heads up, we're missing a link to the original ticket in the PR description.

**Review by kimwoodfield — APPROVED** [review]

**Review by spocke — APPROVED** [review]

**Review by tiny-ben-tran — APPROVED** [review]
I guess that implicit return is not being used for readability.

File: modules/katamari/src/main/ts/ephox/katamari/api/Fun.ts [hunk 2]
```diff
@@ -23,10 +23,20 @@ const constant = <T>(value: T): () => T => {
   };
 };
 
+const constantAsync = <T>(value: T): () => Promise<T> => {
+  return () => {
+    return Promise.resolve(value);
+  };
+};
```
Comment (TheSpyder) [inline modules/katamari/src/main/ts/ephox/katamari/api/Fun.ts:28]: I'm pretty sure this works (and similar for identity async, which is the same function)

```suggestion
const constantAsync = <T>(value: T): () => Promise<T> => {
  return async () => value;
};
```

**Review by TheSpyder — APPROVED** [review]
